### PR TITLE
Factorio: use schema for world_gen, allow setting pollution values

### PR DESCRIPTION
**Pull request by black-sliver**
generate_early: split world_gen into basic and advanced map settings for mod generation

### CODE REVIEW

**Comment by Ijwu** [comment]
What does the `ageing` option do? I can't find any mention of it in the Factorio wiki on the map generation.

**Comment by Berserker66** [comment]
At https://wiki.factorio.com/Prototype/MapSettings it is described as `ageing - double - Constant modifier a percentage of 1; the pollution eaten by a chunks tiles`

**Comment by Ijwu** [comment]
FWIW, this LGTM. I'm not the code owner of this, though, so I'll leave it to @Berserker66.

File: worlds/factorio/Options.py [hunk 1]
```diff
@@ -104,9 +113,57 @@ class FactorioWorldGen(OptionDict):
...
+            Optional("cliff_settings"): {
+                "name": str, "cliff_elevation_0": FloatRange(0,99),
+                "cliff_elevation_interval": FloatRange(0.066,241),  # 40/frequency
+                "richness": FloatRange(0,6)},
+        },
+        "advanced": {
+            Optional("pollution"): {
+                Optional("enabled"): LuaBool,
+                Optional("diffusion_ratio"): FloatRange(0,0.25),
+                Optional("ageing"): FloatRange(0.1,4),
```
Comment (Ijwu) [inline worlds/factorio/Options.py:146]: For this, and other percentage based options, isn't `0` a reasonable minimum? Factorio allows for setting 0%, right?

Comment (black-sliver) [reply worlds/factorio/Options.py:146]: I checked all the values in Factorio GUI when I wrote the code.
Most values seem to be from 1/6 to 6x `FloatRange(0.166,6)`, some have a special 0 (i.e. disable; setting anything 0 and 0.17 will just give you the 17%). Pollution is special in the meaning of its values, so they have different limits in the GUI. Unless i screwed up, the ranges should match the limits in the GUI.

**Comment by Berserker66** [comment]
Thank you!

**Pull request merged**